### DISCUSSION TITLE
Using Vermo for our Identity system

### DISCUSSION THREAD

**Discussion by koptan** [decentralized-identity/veramo#393] [Q&A]
Hi ..

We have our own centralised identity platform and we have a vision to make our platform interoperable with the outside world by using DIDs standard, our system saving Identities and Claims in RDMS .

Our expectation after developing this vision are:

1. Be able to create new DIDs and save it in our DB.
2. Be able to resolve our DIDs by fetching the data from our APIs which are connected to our DB.
3. Be able to resolve SD Request and preparing the SD Response from our APIs.

So my questions are:

1. Can Veramo framework help us in achieving our goal?
2. If yes , then what kind of plugins we need to use or modify or will we have to build new Plugin to achieve the expectations mentioned above?
3. If we are going to create new DID in our system, then should we have new method eg. `did:xxx`? or should we use `did:web` Method Specification?

**Comment by koptan** [comment]
@mirceanis @jasonhealy kindly for your input .

**Comment by mirceanis** [comment]
Hi, yes, that is possible.

1. I believe this framework would be the perfect choice to achieve that.
2. & 3. I think `did:web` would fit the requirements. This is exactly what it was created for; to enable DIDs for the existing web2.0 applications.

To enable resolvers to resolve `did:web:` identifiers hosted by your systems, you would have to expose them as URIs that can serve a `did.json` file.
Please take a look at [the CLI server implementation](https://github.com/uport-project/veramo/blob/16e9339ad8a610d5aff56e01e0eb1faeaf7834f5/packages/cli/src/server.ts#L167) which is an express.js app that uses the [@veramo/remote-server#WebDidDocRouter](https://github.com/uport-project/veramo/blob/16e9339ad8a610d5aff56e01e0eb1faeaf7834f5/packages/remote-server/src/web-did-doc-router.ts#L28) to serve a `did:web` document for a dynamic hostname.

Resolving selective disclosure requests and preparing responses from your APIs should also be possible, but depending on the type of database and the shape of the data this may require the use of custom plugins.
With the [existing set of plugins](https://github.com/uport-project/veramo/tree/16e9339ad8a610d5aff56e01e0eb1faeaf7834f5/packages/data-store), a veramo agent can connect to an SQL database where it can manage both DIDs and verifiable credentials.
Our [selective disclosure protocol implementation](https://github.com/uport-project/veramo/tree/16e9339ad8a610d5aff56e01e0eb1faeaf7834f5/packages/selective-disclosure) can deal with data expressed as atomic verifiable credentials.

Basically, the `did:web` DIDs that you would host, would express some public keys that can be used to verify Verifiable Credential and Presentation signatures. I assume that the credentials would be based on the data in your systems. The `did:web` documents can also list _serviceEndpoints_ that would be used for making selective disclosure requests.

I don't know anything about your data and architecture so I won't venture other explanations, for now, to avoid a potentially confusing response.

**Comment by koptan** [comment]
Thank you so much and sorry for late reply ...
